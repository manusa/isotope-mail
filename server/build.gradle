buildscript {
	ext {
		springBootVersion = '2.0.4.RELEASE'
	}
	repositories {
		mavenCentral()
	}
	dependencies {
		classpath("org.springframework.boot:spring-boot-gradle-plugin:${springBootVersion}")
	}
}

apply plugin: 'java'
apply plugin: 'eclipse'
apply plugin: 'org.springframework.boot'
apply plugin: 'io.spring.dependency-management'

group = 'com.marcnuri.isotope'
version = '0.0.1-SNAPSHOT'
sourceCompatibility = 1.8

repositories {
	mavenCentral()
}


dependencies {
	compile('org.springframework.boot:spring-boot-starter')
	compile('org.springframework.boot:spring-boot-starter-web')
	compile('org.springframework.boot:spring-boot-starter-hateoas')
	compile('com.sun.mail:javax.mail:1.6.1')
	testCompile('org.springframework.boot:spring-boot-starter-test')
	testCompile('com.marcnuri:spring-common-test:0.0.2')
}
