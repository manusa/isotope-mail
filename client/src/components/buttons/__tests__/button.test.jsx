import React from 'react';
import {shallow} from 'enzyme/build/index';
import Button from '../button';

describe('Button component test suite', () => {
  test('Snapshot render, should render button', () => {
    const props = {className: 'first-class', type: 'button', label: 'Black Label',
      icon: 'Andy Warhol', disabled: false};
    const button = shallow(<Button {...props}/>);
    expect(button).toMatchSnapshot();
  });
  test('click, should trigger function', () => {
    // Given
    const onClick = jest.fn();
    const button = shallow(<Button onClick={onClick}/>);

    // When
    button.find('button').simulate('click');

    // Then
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
