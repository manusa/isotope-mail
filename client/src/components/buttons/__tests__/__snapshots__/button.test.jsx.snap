// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component test suite Snapshot render, should render button 1`] = `
<button
  className="mdc-button first-class"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  <i
    className="material-icons mdc-button__icon"
  >
    Andy Warhol
  </i>
  Black Label
</button>
`;
